File: solutions/cpp/log-levels/1/log_levels.cpp
#include <string>

namespace log_line {
    std::string message(std::string line) {
        size_t finder_index = line.find(" ");
        std::string error = line.substr(finder_index + 1);
        return error;
    }

    std::string log_level(std::string line) {
        size_t finder_index = line.find("[");
        size_t end_index = line.find("]");
        std::string error = line.substr(finder_index + 1, end_index - 1);
        return error;
    }

    std::string reformat(std::string line) {
      std::string co_str = log_line::message(line) + " (" + log_line::log_level(line) + ")";
        return co_str;
    }
}

File: solutions/csharp/log-levels/1/LogLevels.cs
using System;
using System.Text.RegularExpressions;

static class LogLine
{
    public static string Message(string logLine)
    {    
        string pattern = @"\s+(.+)";

        string messageSent = Regex.Match(logLine, pattern).Value.Trim();

        return messageSent; 
    }

    public static string LogLevel(string logLine)
    {
        string pattern = @"^.*?\p{P}([a-zA-Z]+)\p{P}.*$";

        string messageLog = Regex.Replace(logLine, pattern, "$1").ToLower();
        
        return messageLog;
    }

    public static string Reformat(string logLine)
    {
        string searchPattern1 = @"\s+(.+)";

        string searchPattern2 = @"^.*?\p{P}([a-zA-Z]+)\p{P}.*$";

        string ccString1 = Regex.Match(logLine, searchPattern1).Value.Trim();

        string ccString2 = Regex.Replace(logLine, searchPattern2, "$1").ToLower();
        
        string newString2 = Regex.Replace(ccString2, @"[^a-zA-Z0-9\s]", "").Trim();
        
        string newMessage = $"{ccString1} ({newString2})";
        //Console.WriteLine($"printing: {newMessage}");

        return newMessage;
    }
}
